2020/Day7.py

```diff
@@ -0,0 +1,99 @@
+# https://adventofcode.com/2020/day/6
+import re
+
+DAY = 7
+
+x_test = '''light red bags contain 1 bright white bag, 2 muted yellow bags.
+dark orange bags contain 3 bright white bags, 4 muted yellow bags.
+bright white bags contain 1 shiny gold bag.
+muted yellow bags contain 2 shiny gold bags, 9 faded blue bags.
+shiny gold bags contain 1 dark olive bag, 2 vibrant plum bags.
+dark olive bags contain 3 faded blue bags, 4 dotted black bags.
+vibrant plum bags contain 5 faded blue bags, 6 dotted black bags.
+faded blue bags contain no other bags.
+dotted black bags contain no other bags.'''.split('\n')
+y_test = 4
+
+y_test2 = 32
+
+
+# Part 1
+def parse_rule(rule_line):
+    bags = re.findall(r'^([\w+ ]+) bags* contain', rule_line)[0]
```

Comment:
I think the `+` in `[\w+ ]` shouldn’t be there—it doesn’t have any special meaning within the brackets, so it will match an actual plus sign. (Which doesn’t matter in practice, but semantically it’s not quite what you want.)

Reply:
That makes sense. I had a surprising amount of trouble with this start of line matching.